src/os/bluestore/BlueStore.cc

```diff
@@ -6322,11 +6323,19 @@ void BlueStore::_txc_state_proc(TransContext *txc)
 		   << dendl;
 	} else {
 	  _txc_finalize_kv(txc, txc->t);
+	  if (g_conf->bluestore_rtc && !txc->is_pipelined_io && txc->wal_txn) {
+	    if (txc->osr->kv_finisher_submitting == 1) {
```

Comment:
Can combine these if's

I still don't understand the relevance of wal_txn here.  You mentioned earlier that it crashed without it.. can you explain what happened?   I don't think we should (need to) treat a wal write any differently than, say, a delete (which has no IO at all).